src/mcedit2/panels/worldmeta.py

```diff
@@ -0,0 +1,255 @@
+"""
+    player
+"""
+from __future__ import absolute_import
+import logging
+from contextlib import contextmanager
+
+from PySide import QtGui
+from PySide.QtCore import Qt
+
+from mcedit2.util.resources import resourcePath
+from mcedit2.util.load_ui import load_ui
+from mcedit2.util.screen import centerWidgetInScreen
+from mcedit2.widgets.nbttree.nbteditor import NBTDataChangeCommand
+
+log = logging.getLogger(__name__)
+
+
+class WorldMetaPanel(QtGui.QWidget):
+    GENERATOR_TYPES = ["default", "flat", "largeBiomes", "amplified", "customized", 'debug_all_block_states']
+    GAMEMODES = ["Survival", "Creative", "Adventure", "Spectator"]
+    DIFFICULTIES = ["Peaceful", "Easy", "Normal", "Hard"]
+
+    editsDisabled = False
+
+    def __init__(self, editorSession):
+        super(WorldMetaPanel, self).__init__(QtGui.qApp.mainWindow, f=Qt.Tool)
+        self.editorSession = editorSession
+        self.worldMeta = editorSession.worldEditor.adapter.metadata
+        callIcon = QtGui.QIcon(resourcePath("mcedit2/assets/mcedit2/icons/edit_metadata.png"))
+        callButton = QtGui.QAction(callIcon, "Edit World Metadata", self)
+        callButton.setCheckable(True)
+        callButton.triggered.connect(self.toggleView)
+        self._toggleViewAction = callButton
+
+        load_ui("panels/worldmeta.ui", baseinstance=self)
+
+        self.worldNBTEditor.editorSession = self.editorSession
+        self.editorSession.revisionChanged.connect(self.revisionDidChange)
+
+        self.updateNBTTree()
+        self.setupPanel()
+        self.updatePanel()
+
+        centerWidgetInScreen(self)
+
+    def updatePanel(self):
```

Comment:
Should probably connect `worldNBTEditor.tagChanged` to `updatePanel` or else update the panel when the QTabWidget's current tab changes like in the Players panel.